tests/test_tools_filter_dom.py

```
@@ -0,0 +1,128 @@
+    assert res["pages"][0]["articles"][0]["headline"] == "Another"
+
+
+def test_filter_dom_file_not_found(tmp_path: Path):
+    with pytest.raises(FileNotFoundError, match="File not found"):
+        filter_dom(tmp_path / "nonexistent.json", tmp_path / "out.json")
+
+
+def test_filter_dom_invalid_input_suffix(tmp_path: Path):
+    invalid = tmp_path / "test.txt"
+    invalid.touch()
+    with pytest.raises(ValueError, match="Input file must be a .json file"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)